adversarial_text/adversarial_losses.py

```diff
@@ -83,19 +83,23 @@ def virtual_adversarial_loss(logits, embedded, inputs,
   """
   # Stop gradient of logits. See https://arxiv.org/abs/1507.00677 for details.
   logits = tf.stop_gradient(logits)
-  # Only care about the KL divergence on the final timestep.
-  weights = _end_of_seq_mask(inputs.labels)
 
+  # Only care about the KL divergence on the final timestep.
+  weights = inputs.eos_weights
+  assert weights is not None
+  
   # Initialize perturbation with random noise.
   # shape(embedded) = (batch_size, num_timesteps, embedding_dim)
-  d = _mask_by_length(tf.random_normal(shape=tf.shape(embedded)), inputs.length)
+  d = tf.random_normal(shape=tf.shape(embedded))
```

Comment:
Just to be clear, is the _mask_by_length here and on line 98 not necessary because the RNN is already parameterized by inputs.length and so won't read the additional entries? Doesn't this make the scale_l2 computation different though since there are more non-zero entries?

And if _mask_by_length is no longer necessary, please remove it.